packages/mui-material/src/Button/Button.js

```diff
@@ -362,13 +510,22 @@ const Button = React.forwardRef(function Button(inProps, ref) {
     endIcon: endIconProp,
     focusVisibleClassName,
     fullWidth = false,
+    id: idProp,
+    loading = false,
+    loadingIndicator: loadingIndicatorProp,
+    loadingPosition = 'center',
     size = 'medium',
     startIcon: startIconProp,
     type,
     variant = 'text',
     ...other
   } = props;
 
+  const id = useId(idProp);
+  const loadingIndicator = loadingIndicatorProp ?? (
+    <CircularProgress aria-labelledby={id} aria-label={`${children}`} color="inherit" size={16} />
```

Comment:
`aria-label={`${children}`}` will produce be equal to `"[object Object]"` if a React element is passed as children. Let's use the exact same implementation that exists in `LoadingButton` for now. See example: https://codesandbox.io/p/sandbox/brave-panini-xm3vsv?file=/src/App.js:4,16

```suggestion
    <CircularProgress aria-labelledby={id} color="inherit" size={16} />
```